chall06/ohakola.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+import sys
+
+# Loops through books and tries to fit them to shelves
+# If count goes too high, return -1
+def shelves_needed(books, shelves, shelf_index, book_index):
+	count = 1
+	curr_shelf = shelves[shelf_index]
+	books_len = len(books)
+	shelves_len = len(shelves)
+	books_placed = 0
+	while books_placed < books_len:
+		if curr_shelf >= books[book_index]:
+			curr_shelf = curr_shelf - books[book_index]
+			book_index += 1
+			if book_index >= books_len:
+				book_index = 0
+			books_placed += 1
+		else:
+			count += 1
+			if count > shelves_len:
```

Comment:
One other pythonic way to approach this would be to create your custom error (e.g. `ShelvesCountError` - I'm terrible at naming), raise it and catch it in the `variable_shelf_pack()` function.